app/manager-frontend/src/components/ConfigureResultsModifierModal.jsx

```diff
@@ -42,8 +43,18 @@ export const ConfirmationModal = ({
   if (!initialData) return null;
   // transform data from the backend so that Type maps to the type id
   // rather than the whole object
+  const getParam = (k, v) => {
```

Comment:
The data that gets sent to the api is in a slightly different format to the data received from the api so I was using this function to transform the initial data to the same structure as the payload data for the api request so that the 2 objects would be easier to compare to one another

Reply:
when sending to the API, are we sending the whole dataset? or individual records at a time?

If the latter, do the transformation at the point of sending on an individual record.

Or could we change the backend so they're closer to each other? (May or may not fit the spirit of each case, but worth a look).

Also (related), why stringify a JSON object and repeatedly parse the string later (for display, for posting, etc...) We're in JS, why wouldn't we use native objects?!

Happy to have an actual call or something on this for the context, but currently it doesn't look like this is the sensible way round to do it.

Reply:
Changed the code so that the transformation is done at the point of sending an individual record. The reason I was stringifying JSON objects was because the value field in a <select> <option value={val}> would only take a string. The way I have changed formikSelect now has removed the need for that tho so it now just uses native objects.